v2/piv/piv.go

```diff
@@ -364,14 +369,39 @@ var (
 	aidYubiKey    = [...]byte{0xa0, 0x00, 0x00, 0x05, 0x27, 0x20, 0x01, 0x01}
 )
 
-func ykAuthenticate(tx *scTx, key [24]byte, rand io.Reader) error {
+func ykAuthenticate(tx *scTx, key []byte, rand io.Reader, version *version) error {
 	// https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-73-4.pdf#page=92
 	// https://tsapps.nist.gov/publication/get_pdf.cfm?pub_id=918402#page=114
 
+	var managementKeyType byte
+	if supportsVersion(version, 5, 3, 0) {
+		// if yubikey version >= 5.3.0, determine management key type using slot metadata
+		cmd := apdu{
+			instruction: insGetMetadata,
+			param1:      0x00,
+			param2:      keyCardManagement,
+		}
+		resp, err := tx.Transmit(cmd)
+		if err == nil {
+			managementKeyType = resp[2:][0]
+		}
+	}
+
+	// set challengeLength based on managementKeyType
+	var challengeLength byte
+	switch managementKeyType {
```

Comment:
What happens if this doesn't match the length of the key? I'd imagine we'd want to immediately bail out.

Reply:
This is already handled with a reasonable error message by crypto:
![Screenshot 2024-07-08 at 16 09 50](https://github.com/go-piv/piv-go/assets/673216/249846a1-8d52-4f58-ab88-b3bedd651e76)
![Screenshot 2024-07-08 at 16 10 03](https://github.com/go-piv/piv-go/assets/673216/a7469d45-0a35-4992-baad-24f3cb47fb20)

Reply:
Added validation and slightly more useful error message.